init.el

```diff
@@ -37,6 +37,12 @@ straight.el or Guix depending on the value of
         (message "Package '%s' does not appear to be installed by Guix!"))
     (straight-use-package package)))
 
+;; Check the system used
+(defconst ON-LINUX   (eq system-type 'gnu/linux))
```

Comment:
Since the type of OS can only ever be one of these options at a time, perhaps we should boil it down to a single variable?  Maybe something like `rational-current-operating-system` which has the values of `windows`, `macos`, or `linux`?

Reply:
In my configuration I have all those operating systems, but I also check for graphical system, like this.

      (memq window-system '(mac ns x))

And I uses `darwin` to set my keyboard for my Mac, that is just OS, no GUI related.